components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/bean/RuleTemplate.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.business.rules.core.bean;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Represents a Rule Template, which consists of one or more Templates
+ */
+public class RuleTemplate {
+    private String uuid;
+    private String name;
+    private String type;
+    private String instanceCount;
+    private String script;
+    private String description;
+    private Collection<Template> templates;
+    private Map<String, RuleTemplateProperty> properties; // Name, RuleTemplateProperty object
+
+    public RuleTemplate(String uuid, String name, String type, String instanceCount, String script, String description,
+                        Collection<Template> templates, Map<String, RuleTemplateProperty> properties) {
```

Comment:
Why get `RuleTemplateProperty`s as a `Map` instead of a `Set`? Don't ask for a `Map`, the caller might fill the map with different type of string as the key. Ask for a `Set` and convert that into a `Map` inside the constructor. e.g. https://github.com/wso2/carbon-ui-server/blob/v0.10.4/components/org.wso2.carbon.uis/src/main/java/org/wso2/carbon/uis/api/App.java#L69

Reply:
This is used as a bean class. It will be instantiated with gson. That's why we get a map directly to the constructor. If there is a malformed json, it will be failed with a proper log where conversion is done.

Reply:
ok